pkg/kubelet/status/status_manager.go

```diff
@@ -683,20 +683,35 @@ type SidecarsStatus struct {
 
 // GetSidecarsStatus returns the SidecarsStatus for the given pod
 func GetSidecarsStatus(pod *v1.Pod) SidecarsStatus {
+	var containerStatusesCopy []v1.ContainerStatus
 	if pod == nil {
 		klog.Infof("Pod was nil, returning empty sidecar status")
 		return SidecarsStatus{}
 	}
-	if pod.Spec.Containers == nil || pod.Status.ContainerStatuses == nil {
-		klog.Infof("Pod Containers or Container status was nil, returning empty sidecar status")
+	if pod.Spec.Containers == nil {
+		klog.Infof("Pod Containers was nil, returning empty sidecar status")
 		return SidecarsStatus{}
 	}
```

Comment:
We could consider being even more defensive and returning `sidecarsPresent:true` + `sidecarsReady:false` in these 2 `if`s.